pkg/cmd/gist/view/view.go

```diff
@@ -93,12 +94,22 @@ func viewRun(opts *ViewOptions) error {
 
 	theme := opts.IO.DetectTerminalTheme()
 	markdownStyle := markdown.GetStyle(theme)
-	if err := opts.IO.StartPager(); err != nil {
-		fmt.Fprintf(opts.IO.ErrOut, "starting pager failed: %v\n", err)
-	}
-	defer opts.IO.StopPager()
 
 	render := func(gf *shared.GistFile) error {
+		isBinary, err := shared.DetectedBinaryFileContents(gf.Content, true)
```

Comment:
When reading an existing gist, every GistFile returned from the API already reports its mime-type in the `type` field. We can trust this mime-type and avoid detecting one from file contents.

Reply:
You're right. But then to determine whether or not a file is binary, we need to maintain a list of binary mime types to block all of them(or a list to non-binary types to proceed). On another note [mimetype](https://github.com/gabriel-vasile/mimetype) detects the mime of a file only if it is given a physical file on the local machine or its contents and not its extension or file-type(for eg: `image/png`). That's why I chose to pass the contents of the file itself. 

Moreover [mimetype](https://github.com/gabriel-vasile/mimetype) only reads 3072 bytes of the file, so I thought it wouldn't be a big deal. 

Would like to know your thought process on this too.